dali/kernels/common/join/tensor_join_cpu.h

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_TENSOR_JOIN_CPU_H
+#define DALI_TENSOR_JOIN_CPU_H
+
+#include "dali/kernels/kernel.h"
+#include "dali/core/tensor_shape_print.h"
+#include "dali/core/format.h"
+
+namespace dali {
+namespace kernels {
+
+enum JoinMode {
+  STACK,
+  CONCAT
+};
+
+namespace detail {
+
+template<typename T>
+void TransferBuffers(span<T> output, const TensorShape<> &output_shape,
```

Comment:
The name is generic to the point of being meaningless. This function actually... concatenates tensors, so the name should read 
```suggestion
void ConcatenateTensors(span<T> output, const TensorShape<> &output_shape,
```

Reply:
BTW - why is output split into shape and buffer?